### PR TITLE
Merge in btcd commit 2adfb3b56acd280e84451e94dd0c06203eef9832

### PR DESCRIPTION
**Pull request by cjepson**
Merges in btcd commit 2adfb3b56acd280e84451e94dd0c06203eef9832.

### CODE REVIEW

**Comment by alexlyp** [comment]
tested on simnet and works as expected

File: wire/msggetheaders.go [hunk 1]
```diff
@@ -65,14 +65,16 @@ func (msg *MsgGetHeaders) BtcDecode(r io.Reader, pver uint32) error {
 		return messageError("MsgGetHeaders.BtcDecode", str)
 	}
 
+	// reduce the number of allocations.
```
Comment (davecgh) [inline wire/msggetheaders.go]: Missing part of the comment from upstream.

```// Create a contiguous slice of hashes to deserialize into in order to```

**Comment by cjepson** [comment]
@davecgh I agree, I'll add an issue for the merkleblock stuff.

File: wire/msggetheaders.go (hunk 1 above)
Comment (cjepson) [inline wire/msggetheaders.go]: Added back

**Comment by cjepson** [comment]
@davecgh I have added the optimization for stake transactions as well in msgmerkleblock.go.

**Pull request merged**